templates/terraform/resource.erb

```diff
@@ -261,17 +251,17 @@ func resource<%= resource_name -%>Read(d *schema.ResourceData, meta interface{})
 <%  end -%>
 
 <%  object.gettable_properties.reject{|p| p.ignore_read }.each do |prop| -%>
+<%# Terraform must set the top level schema field, but with flattened properties it's difficult to
+know what the top level should be. Instead we just loop over the map returned from flatten. -%>
 <%    if prop.flatten_object -%>
-    if v, ok := res["<%= prop.api_name -%>"].(map[string]interface{}); res["<%= prop.api_name -%>"] != nil && ok {
-<%      prop.properties.each do |p| -%>
-      if err := d.Set("<%= p.name.underscore -%>", flatten<%= resource_name -%><%= titlelize_property(prop) -%><%= titlelize_property(p) -%>(v["<%= p.api_name -%>"], d)); err != nil {
-        return fmt.Errorf("Error reading <%= object.name -%>: %s", err)
-    }
-<%      end -%>
-    } else {
-<%      prop.properties.each do |p| -%>
-      d.Set("<%= p.name.underscore -%>", nil)
-<%      end -%>
+    // Handle flattened_object
```

Comment:
Can we move the ERB comment into the generated code? Providing context there is helpful for readers in the future.